### PR TITLE
Add New 3D Incompressible Channel Flow Test and Unstructured Mesh Reader 

### PR DESCRIPTION
**Pull request by Yashmandaokar**
3D Unstructured Mesh Reader (test_3d_unstructured_mesh_reader):
Extends the existing mesh reader capabilities to support 3D meshes created from ICEM and Fluent.

3D Incompressible Inviscid Channel Flow Test (test_3d_incompressible_channel_flow):
Provides an inviscid channel flow example

### CODE REVIEW

**Comment by Xiangyu-Hu** [comment]
@Yashmandaokar the issue is that the Code need to be compiled in Linux.

**Comment by Yashmandaokar** [comment]
> @Yashmandaokar the issue is that the Code need to be compiled in Linux.

@Xiangyu-Hu. 
Thank you for bringing this to my attention. Could you kindly advise me on the best approach to address this issue?
 Should I focus on resolving the errors indicated in the Linux build log, or would it be more effective to attempt compiling the code using a gcc or clang compiler? 
Your guidance would be greatly appreciated.

**Comment by Xiangyu-Hu** [comment]
Yes. There should small issues. Usually due to more strict rules in linux.

**Comment by Yashmandaokar** [comment]
@Xiangyu-Hu I have rectified the errors. Now the code is compiling on linux. Could you please run the build and test.

**Comment by Xiangyu-Hu** [comment]
ok

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,9 +4,7 @@ name: CI
 on:
   # Triggers the workflow on push or pull request events but only for the master branch
   push:
-    branches: [master]
-  pull_request:
-    branches: [master]
+    branches: [Incompressible-taper-channel-flow]
```
Comment (Xiangyu-Hu) [inline .github/workflows/ci.yml:7]: The CI from the main branch should not be changed.

**Review by Xiangyu-Hu — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Xiangyu-Hu) [reply .github/workflows/ci.yml:7]: I mean the entire ci.yml file need to be changed back.

Comment (Yashmandaokar) [reply .github/workflows/ci.yml]: Sorry I changed the file to test my code for linux build. I have now changed the Ci.yml file back to the original.

**Comment by FengWang3119** [comment]
I am curious why there is a class called wavebody in test_3d_incompressible_channel_flow.h. Is it a straight channel?

**Comment by Xiangyu-Hu** [comment]
> I am curious why there is a class called wavebody in test_3d_incompressible_channel_flow.h. Is it a straight channel?

I agree with the comment. The naming needs to be expressive and mean what it should.

**Comment by Yashmandaokar** [comment]
> > I am curious why there is a class called wavebody in test_3d_incompressible_channel_flow.h. Is it a straight channel?
> 
> I agree with the comment. The naming needs to be expressive and mean what it should.

I will make the namings expressive and meaningfull. Thank you for your feedback.

File: src/shared/bodies/complex_bodies/unstructured_mesh_3d.h
Comment (Xiangyu-Hu) [inline src/shared/bodies/complex_bodies/unstructured_mesh_3d.h:1]: Could 2d and 3d unstructured meshes share the same header? I found that this file is very similar to the header for 2d mesh.

File: src/shared/bodies/complex_bodies/MeshHelper.h
Comment (Xiangyu-Hu) [inline src/shared/bodies/complex_bodies/MeshHelper.h:1]: Will this helpler is also working for 2d cases?

File: src/shared/bodies/complex_bodies/MeshHelper.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
+/* -------------------------------------------------------------------------*
+ *								SPHinXsys									*
+ * -------------------------------------------------------------------------*
+ * SPHinXsys (pronunciation: s'finksis) is an acronym from Smoothed Particle*
+ * Hydrodynamics for industrial compleX systems. It provides C++ APIs for	*
+ * physical accurate simulation and aims to model coupled industrial dynamic*
+ * systems including fluid, solid, multi-body dynamics and beyond with SPH	*
+ * (smoothed particle hydrodynamics), a meshless computational method using	*
```
Comment (Xiangyu-Hu) [inline src/shared/bodies/complex_bodies/MeshHelper.h:8]: You can copy the credit claims from any header of the master, where there is no tab used.

File: src/shared/bodies/complex_bodies/unstructured_mesh_3d.h
Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/unstructured_mesh_3d.h:1]: Yes they can share the same header.

File: src/shared/bodies/complex_bodies/MeshHelper.h
Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/MeshHelper.h:1]: With minor modifications, some of the functions from MeshHelper.h can be used directly. However, most of the functions would need to be adapted for the 2D case.

File: src/shared/bodies/complex_bodies/MeshHelper.h (hunk 2 above)
Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/MeshHelper.h:8]: On my local machine it looks fine. However when I push it on the repository it gets distorted. I will change it.

Screenshot 2024-04-09 075639

**Review by Lisu-lisa — APPROVED** [review]

File: src/shared/bodies/complex_bodies/MeshHelper.h (hunk 2 above)
Comment (Xiangyu-Hu) [reply src/shared/bodies/complex_bodies/MeshHelper.h:8]: The reason is due to the tab which is usually defined differently by different systems. The new version in master does not used any tab.

File: src/shared/bodies/complex_bodies/MeshHelper.h
Comment (Xiangyu-Hu) [reply src/shared/bodies/complex_bodies/MeshHelper.h:1]: I mean that they can share the same function name but can be implemented differently in 2 and 3d.

Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/MeshHelper.h:1]: Yes that can be done. Should I do it?

Would it be possible to stop or cancle the ci workflow? As the latest commit only has credit claim changed.

Comment (Xiangyu-Hu) [reply src/shared/bodies/complex_bodies/MeshHelper.h:1]: Yes. You can cancel the present CI or simply do not mind it.

File: src/shared/bodies/complex_bodies/MeshHelper.h (hunk 2 above)
Comment (DrChiZhang) [reply src/shared/bodies/complex_bodies/MeshHelper.h]: For file name, mesh_helper.h and mesh_helper.cpp would be better.

Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/MeshHelper.h]: In the latest commit I have updated the file name as suggested. Thank you for the feedback.

File: src/shared/bodies/complex_bodies/unstructured_mesh_3d.h
Comment (Yashmandaokar) [reply src/shared/bodies/complex_bodies/unstructured_mesh_3d.h:1]: I have made the changes suggested by you. Now the header file for mesh_helper and unstructured mesh is same. Thank you for the review and sugesstion.

**Pull request merged**